exe/genv-envs

```diff
@@ -17,17 +17,27 @@ QUERIES = {
     'config.gpu_memory': lambda env: env["config"].get("gpu_memory"),
 }
 
+
 def factory() -> Dict:
+    """
+    Create a 'empty' state object for genv environments.
+    :return: The state object (dict of dicts)
+    """
     return dict(
         envs={}
     )
 
+
 def cleanup(
         envs: Envs,
-        poll_pid: Callable[[int], bool]=genv.poll_pid,
-        poll_kernel: Callable[[str], bool]=genv.poll_kernel) -> None:
+        poll_pid: Callable[[int], bool] = genv.poll_pid,
+        poll_kernel: Callable[[str], bool] = genv.poll_jupyter_kernel) -> None:
     """
-    Clean up processes, kernels and environments.
+    Clean up processes, kernels and environments connected to the given environment.
+    :param envs: dictionary describing all existing environments.
+    :param poll_pid: a predicate which tells us which pids are still connected to the environment.
+    :param poll_kernel: a predicate which tells us which jupyter kernels are still connected to the environment.
```

Comment:
I think "connected to the environment" should be replaced with "running"

Reply:
See my answer on my previos comment.

Reply:
I changed it to "an environment". This implies that we work on proccesses and kernels from any environments.